crates/proto/src/https/https_client_stream.rs

```diff
@@ -733,11 +734,32 @@ mod tests {
         );
     }
 
-    fn client_config_tls12_webpki_roots() -> ClientConfig {
-        use rustls::{OwnedTrustAnchor, RootCertStore};
+    fn client_config_tls12() -> ClientConfig {
+        use rustls::RootCertStore;
+        #[cfg_attr(
+            not(any(feature = "native-certs", feature = "webpki-roots")),
+            allow(unused_mut)
+        )]
         let mut root_store = RootCertStore::empty();
+        #[cfg(all(feature = "native-certs", not(feature = "webpki-roots")))]
+        {
+            let (added, ignored) = root_store
+                .add_parsable_certificates(&rustls_native_certs::load_native_certs().unwrap());
+
+            if ignored > 0 {
+                warn!(
+                    "failed to parse {} certificate(s) from the native root store",
+                    ignored
+                );
+            }
+
+            if added == 0 {
+                panic!("no valid certificates found in the native root store");
```

Comment:
we should change the result type of this function to `Result<ClientConfig, ...>` and return an error here. We don't want to panic unless absolutely necessary.

Reply:
IIRC this is for tests only, so panicking is fine.

Reply:
oh, you're right. It's in the test module, I should have expanded the review more. Thanks for pointing that out, @djc